cuda/ubi8-python-3.8/Dockerfile

```diff
@@ -0,0 +1,86 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+LABEL name="odh-notebook-cuda-ubi8-python-3.8" \
+      summary="CUDA Python 3.8 base image for ODH notebooks" \
+      description="CUDA Python 3.8 builder image based on UBI8 for ODH notebooks" \
+      io.k8s.display-name="CUDA Python 3.8 base image for ODH notebooks" \
+      io.k8s.description="CUDA Python 3.8 builder image based on UBI8 for ODH notebooks" \
+      authoritative-source-url="https://github.com/opendatahub-io/notebooks" \
+      io.openshift.build.commit.ref="main" \
+      io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/cuda/ubi8-python-3.8" \
+      io.openshift.build.image="quay.io/opendatahub/notebooks:cuda-ubi8-python-3.8"
+
+# Install CUDA base from:
+# https://gitlab.com/nvidia/container-images/cuda/-/blob/master/dist/11.4.2/ubi8/base/Dockerfile
+USER 0
+WORKDIR /opt/app-root/bin
+
+ENV NVARCH x86_64
+ENV NVIDIA_REQUIRE_CUDA "cuda>=11.4 brand=tesla,driver>=418,driver<419 brand=tesla,driver>=450,driver<451"
+ENV NV_CUDA_CUDART_VERSION 11.4.108-1
+
+COPY cuda.repo-x86_64 /etc/yum.repos.d/cuda.repo
+
+RUN NVIDIA_GPGKEY_SUM=d0664fbbdb8c32356d45de36c5984617217b2d0bef41b93ccecd326ba3b80c87 && \
+    curl -fsSL https://developer.download.nvidia.com/compute/cuda/repos/rhel8/${NVARCH}/D42D0685.pub | sed '/^Version/d' > /etc/pki/rpm-gpg/RPM-GPG-KEY-NVIDIA && \
+    echo "$NVIDIA_GPGKEY_SUM  /etc/pki/rpm-gpg/RPM-GPG-KEY-NVIDIA" | sha256sum -c --strict -
+
+ENV CUDA_VERSION 11.4.2
```

Comment:
11.4 or 11.6? According to [this](https://access.redhat.com/documentation/en-us/red_hat_openshift_data_science/1/html/getting_started_with_red_hat_openshift_data_science/launching-jupyter-and-starting-a-notebook-server_get-started#options-for-notebook-server-environments_get-started), we should use the 11.6

Reply:
Base image is `11.4.2` -> https://github.com/red-hat-data-services/odh-deployer/blob/main/nbc/cuda-11.4.2/manifests.yaml#L16

The docs say 11.6 because of this issue -> https://issues.redhat.com/browse/RHODS-4312